File: 38-count-and-say/count-and-say.cpp
class Solution {
public:
    string rle(string s)
    {
        string res;
        int count=1;
        for(int i=0;i<s.size();i++)
        {
            if(s[i]==s[i+1]){count++;}
            else
            {
                string t=to_string(count);
                for(int j=0;j<t.size();j++)
                {
                    res.push_back(t[j]);
                }
                res.push_back(s[i]);
                count=1;
            }     
        }
        //string t=to_string(count);
        //for(int j=0;j<t.size();j++)
              //  {
                 //   res.push_back(t[j]);
               // }
       // res.push_back(s[s.size()-1]);
        return res;
    }

    string countAndSay(int n) {
        if(n==1)return "1";

        return rle(countAndSay(n-1));
    }
};

File: 38-count-and-say/count-and-say.java
class Solution {
    String res="";
    public String countAndSay(int n) {
        return solve(n);
    }
    public String solve(int n){
        if(n==1){
            res="1";
            return res;
        }
        String curr=solve(n-1);
        StringBuilder sb=new StringBuilder("");
        int count=1;
        for(int i=0;i<curr.length();i++){
            if(i+1<curr.length()&&curr.charAt(i)==curr.charAt(i+1)) count++;
            else{
                sb.append(count);
                sb.append(curr.charAt(i));
                count=1;
            }
        }
        return res=sb.toString();
    }
}